### ISSUE TITLE
TextAlign([any], CENTER) does not forcibly wrap text inside of the text block, and the text will spill outside of it 

### ISSUE THREAD

**Issue by trifoilarity** [processing/p5.js#5785]
### Most appropriate sub-area of p5.js?

- [ ] Accessibility
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [X] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Build Process
- [ ] Unit Testing
- [ ] Internalization
- [ ] Friendly Errors
- [ ] Other (specify if possible)

### p5.js version

1.4.2

### Web browser and version

Version 1.43.88 Chromium: 105.0.5195.68 (Official Build) (64-bit)

### Operating System

Ubunto 20.04.1

### Steps:
1. use textAlign([any], CENTER)
2. Create a long string and put it in a text box with a x,y,w,h limit defined
3. It will render outside of the box

### Snippet:

```
textAlign(CENTER, CENTER)
text("Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World Hello World", 100, 100, 200, 200)
```

![spilling-text-box](https://user-images.githubusercontent.com/737638/188300741-716c3ec8-78c4-4b87-afec-466f55f8874a.png)

**Comment by davepagurek** [comment]
I wonder if the screenshot in this issue is actually the expected result: text wrapping is just based on the text width (we don't try to break text into columns to fit vertically) and the text is too long to fit in the box, so this seems like the correct output.

If I wanted to clip to the box, there are alternate methods one can use (maybe these can be more easily accessible?) such as `drawingContext.clip()`:

<table>
<tr>
<td>

```js
push()
fill(0, 20)
rect(0, 0, 100, 100)
drawingContext.clip()
  
fill(0)
text(longStringHere, 0, 0, 100, 100)
pop()
```

</td>
<td>
<img src="https://user-images.githubusercontent.com/5315059/188285199-54c26777-411f-4c51-898d-d1d0233912ca.png">
</td>
</tr>
</table>

**Comment by dhowe** [comment]
I think the basic issue here is that the alignments are not working consistently when a bounding box is specified (and the behavior does not match the docs):

CENTER/TOP:
<img width="238" alt="image" src="https://user-images.githubusercontent.com/737638/188301236-5ad57ec6-ee01-4a09-9779-962ea42ebfff.png">

CENTER/CENTER:
<img width="222" alt="image" src="https://user-images.githubusercontent.com/737638/188301264-82a58ffb-220b-4b2d-8032-9ee78043890f.png">

CENTER/BOTTOM:
<img width="225" alt="image" src="https://user-images.githubusercontent.com/737638/188301276-854191e3-9317-4684-b141-6f0e130e4916.png">

**Comment by Qianqianye** [comment]
Thanks @trifoilarity @dhowe @davepagurek. It's fixed in #5787.

**Comment by dhowe** [comment]
@Qianqianye I'm not sure if we have a unified way of handling warnings, but in the case of text(), the second two parameters are each specified as optional, but it is not clear from the docs that in _some_ cases the 4th is required when the 3rd is included. In the PR linked above, I've added warnings for such cases, but I'm not sure that this is the best approach...

For example:
```js
textAlign(TOP, TOP);
text('Ok',  10, 10, 200);
```

```js
textAlign(TOP, CENTER);
text('Not Ok',  10, 10, 200);
```
